state.c

```diff
@@ -444,6 +444,8 @@ int pv_state_validate(struct pv_state *s)
 
 static int pv_state_mount_bsp_volumes(struct pv_state *s)
 {
+	pv_disk_mount_all(s);
```

Comment:
I think this kind of thing would break non-reboot updates.

pv_state_mount_bsp_volumes is about mounting the BSP volumes. That means mounting every volume that is not linked to a platform, such as the metadata volumes. It is called from pv_state_start, which is called from _pv_run in pantavisor.c and it is only called when we are not transitioning (non-reboot updates).

For disks, we had relied on the disk pointer that we store in each volume. When a platform is being started, we go through the list of volumes and call pv_volume_mount for the disks that are associated to each platform. Then, inside of pv_volume_mount we mount the disk if necessary with the handler, which was the only way to actually. And that's the only place we should be mounting disks, if they are required by a platform, that way, you can add a new platform that uses a new disk and everything work fine, because the work is done at the platform level basically. Same with umount.

Reply:
ack. If we are going to only mount swap disks in this point, I'd suggest to do it in a diff function named pv_state_mount_bsp_disks that is called before this one in pv_state_start